rflx/converter/iana.py

```diff
@@ -0,0 +1,279 @@
+            location=Location(start=(0, 0), source=source),
+        )
+    package_name = _normalize_name(registry_id)
+    registry_last_updated = root.find("iana:updated", NAMESPACE)
+    registry_title = root.find("iana:title", NAMESPACE)
+
+    enum_types = []
+    for registry in root.findall(root.tag):
+        enum_type = _convert_registry_to_enum_type(registry, always_valid)
+        if enum_type is not None:
+            enum_types.append(enum_type)
+    resolve_duplicate_literals(enum_types)
+    file = output_dir / Path(file_name(package_name) + ".rflx")
+    write_rflx_specification(
+        file, enum_types, package_name, registry_title, registry_last_updated, reproducible
+    )
+
+
+def write_rflx_specification(
+    file: Path,
+    enum_types: List[EnumType],
+    package_name: str,
+    registry_title: Optional[Element],
+    registry_last_updated: Optional[Element],
+    reproducible: bool,
+) -> None:
+    with open(file, "w+", encoding="utf-8") as f:
+        f.write(
+            "-- AUTOMATICALLY GENERATED. DO NOT EDIT.\n"
+            + (
+                ""
+                if reproducible
+                else f"-- Generation date: {datetime.now().strftime('%Y-%m-%d')}\n"
+            )
+        )
+        if registry_title is not None:
+            f.write(f"-- {registry_title.text}\n")
+        if registry_last_updated is not None:
+            f.write(f"-- Registry last updated on {registry_last_updated.text}\n\n")
+        f.write(f"package {package_name} is\n\n")
+        for enum_type in enum_types:
+            f.write(str(enum_type))
+        f.write(f"end {package_name};\n")
+
+
+def resolve_duplicate_literals(enum_types: List[EnumType]) -> None:
+    literal_names = [
+        enum_literal.name for enum_type in enum_types for enum_literal in enum_type.enum_literals
+    ]
+    for enum_type in enum_types:
+        for enum_literal in enum_type.enum_literals:
+            if literal_names.count(enum_literal.name) > 1:
+                enum_literal.name += f"_{enum_literal.value.replace('16#', '').replace('#', '')}"
+
+
+def _convert_registry_to_enum_type(
+    registry: Element,
+    always_valid: bool,
+) -> Optional[EnumType]:
+    # pylint: disable=too-many-locals
+    records = registry.findall("iana:record", NAMESPACE)
+    title = registry.find("iana:title", NAMESPACE)
+    if len(records) == 0 or title is None or title.text is None:
+        return None
+
+    registry_title = _normalize_name(title.text)
+    enum_literals: Dict[str, EnumLiteral] = {}
+    enum_literal_highest_bit_len = 0
+    for record in records:
+        name_tag = _get_name_tag(record)
+        value_tag = "value"
+
+        if name_tag is None:
+            continue
+
+        name_element = record.find(f"iana:{name_tag}", NAMESPACE)
+        value_element = record.find(f"iana:{value_tag}", NAMESPACE)
+
+        if value_element is None or name_element is None:
+            continue  # pragma: no cover
```

Comment:
Why is this pragma needed here? Is that a similar case as https://github.com/Componolit/RecordFlux/blob/main/rflx/specification/style.py#L173-L179? If so, I think we should rather use the same workaround and note here.

Reply:
I wasn't aware of that issue / fix. Adopted it.